src/main/java/org/team1540/robot2023/commands/drivetrain/GridDriveCommand.java

```diff
@@ -6,26 +6,40 @@
 import com.pathplanner.lib.PathPoint;
 import com.pathplanner.lib.server.PathPlannerServer;
 import edu.wpi.first.apriltag.AprilTag;
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.math.geometry.Pose2d;
 import edu.wpi.first.math.geometry.Rotation2d;
 import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
 import org.team1540.robot2023.Constants;
 import org.team1540.robot2023.utils.PolePosition;
 
 import static org.team1540.robot2023.Globals.aprilTagLayout;
+import static org.team1540.robot2023.Globals.field2d;
 
 class GridDriveCommand extends SequentialCommandGroup {
     private static int getClosestTag(Drivetrain drivetrain) {
+        aprilTagLayout.setOrigin(DriverStation.getAlliance() == DriverStation.Alliance.Red ? AprilTagFieldLayout.OriginPosition.kRedAllianceWallRightSide : AprilTagFieldLayout.OriginPosition.kBlueAllianceWallRightSide);
         Translation2d currentPose = drivetrain.getPose().getTranslation();
         double mindist = Double.MAX_VALUE;
         int closestTag = -1;
-        for (AprilTag tag : aprilTagLayout.getTags()) {
-            double distance = tag.pose.toPose2d().getTranslation().getDistance(currentPose);
-            if (distance < mindist || closestTag == -1) {
-                mindist = distance;
-                closestTag = tag.ID;
+        try {
+            for (AprilTag tag : aprilTagLayout.getTags()) {
+                Pose2d pose = aprilTagLayout.getTagPose(tag.ID).orElseThrow().toPose2d();
+                double distance = pose.getTranslation().getDistance(currentPose);
+                field2d.getObject("apriltag/" + tag.ID).setPose(pose);
+                if (distance < mindist || closestTag == -1) {
+                    mindist = distance;
+
+                    closestTag = tag.ID;
+                }
             }
+        }  catch (Exception e) {
```

Comment:
Does this throw a generic Exception or does it actually throw some helpful exception? If it throws a specific one when failing catch that one instead of all of the exceptions.